### ISSUE TITLE
Build with clang broken on Linux

### ISSUE THREAD

**Issue by josephbisch** [goatpig/BitcoinArmory#137]
I get the following error when trying to build Armory from source on Linux using clang.

```
make -C cppForSwig
make[1]: Entering directory '/home/joseph/bitcoinarmory/BitcoinArmory/cppForSwig'
clang++ -Icryptopp -Imdb -IBlockDataManager/fcgi -DUSE_CRYPTOPP -D__STDC_LIMIT_MACROS -I/usr/include/python2.7 -I/usr/include/python2.7 -std=c++11 -O2 -pipe -fPIC -c BDM_mainthread.cpp
In file included from BDM_mainthread.cpp:14:
In file included from ./BDM_mainthread.h:23:
In file included from ./BDM_Server.h:23:
./DataObject.h:150:4: error: constructor cannot be redeclared
   IntType(size_t val) : val_(val)
   ^
./DataObject.h:132:4: note: previous declaration is here
   IntType(uint64_t val) : val_(val)
   ^
1 error generated.
make[1]: *** [Makefile:116: BDM_mainthread.o] Error 1
make[1]: Leaving directory '/home/joseph/bitcoinarmory/BitcoinArmory/cppForSwig'
make: *** [Makefile:9: all] Error 2
```

I guess the second declaration is necessary for OS X builds and that is why it is there?

**Comment by goatpig** [comment]
size_t is always arch dependent. This is what you have to use to build in x86 and x64 without pointers blowing up and offsets overflowing.

GCC and MSVC keep their x64 declaration different as well, although the result is the same: 64 bit unsigned integer.

The best to do in the OSX case would be to guard the uint64_t declaration with #ifdef.

**Comment by droark** [comment]
This should be fixed by #148.

**Comment by achow101** [comment]
#148 has been merged. This can be closed now.
